### DISCUSSION TITLE
Suggestions on modifying the Driver Order status to truly reflect the vehicle execution status

### DISCUSSION THREAD

**Discussion by BenYaho** [openTCS/opentcs#146] [Q&A] [answered]
Dear OpenTCS Team,

I am inquiring about any recommended modifications to align the status changes of DriverOrder with the actual execution status of the vehicle, specifically encompassing states such as TRAVELLING, OPERATING, FINISHED, and FAILED.

Thank you for your attention to this request, and I look forward to hearing from you soon.

**Comment by swltr** [comment]
@BenYaho It's not quite clear what you mean here. There are multiple states:

* The vehicle's state is reported by the vehicle driver.
* The drive order's state depends on the progress made while processing it.

What is it you're trying to achieve?

**Reply by BenYaho** [reply]
Hi @swltr, 

I want to achieve the effect where the drive order's state depends on the progress made while processing it. 
Currently, I am using version 5.17.0. I tried to set the drive order's state using order.withCurrentDriveOrderState(DriveOrder.State.OPERATING), but it didn't work.
Do you have any suggestions on how to set the drive order's state properly?

**Reply by swltr** [reply, marked as answer]
@BenYaho As I wrote in my previous message: the drive order's state _does_ depend on the progress. The intended flow is like this:

```mermaid
flowchart LR
  PRISTINE --> TRAVELLING --> FINISHED
  TRAVELLING --> FAILED
```

This is what is implemented in the baseline code, and what you should see as long as the vehicle driver properly reports back completion of movement commands.

`OPERATING` is a remnant that is not really used any more and may eventually be removed.
